core/src/helpers/parseMetamaskError.ts

```diff
@@ -6,9 +7,16 @@ const parseMetamaskError = function (errorMessage = ''): string {
     try {
         const jsonString = errorMessage.substring(jsonFirstIndex, jsonLastIndex + 1);
         const metamaskError = JSON.parse(jsonString);
-        return truncateText(metamaskError?.value?.data?.message || 'unknown');
+
+        const rawErrorMessage = metamaskError?.value?.data?.message || 'unknown';
+        const parsedErrorMessage = METAMASK_ERRORS[rawErrorMessage] || rawErrorMessage;
```

Comment:
This issue is about improving how we parse error messages! We don't have to store a list or exact strings that should display something else instead

Reply:
In other words, you need to change the logic above those lines, where we try to `JSON.parse` the message based on the hacked together `jsonFirstIndex, jsonLastIndex` which doesn't always work as it seems!